packages/create-app/src/lib/tasks.ts

```diff
@@ -236,3 +236,59 @@ export async function moveAppTask(
       });
   });
 }
+
+/**
+ * Checks if git package is installed and git credentials exists
+ *
+ * @throws if `exec` fails
+ */
+export async function checkForGitSetup(): Promise<boolean> {
+  const runCmd = (cmd: string) =>
+    exec(cmd).catch(error => {
+      process.stdout.write(error.stderr);
+      process.stdout.write(error.stdout);
+      throw new Error(`Could not execute command ${chalk.cyan(cmd)}`);
+    });
+
+  try {
+    await runCmd('which git');
+
+    const [gitUsername, gitEmail] = await Promise.all([
+      runCmd('git config user.name'),
+      runCmd('git config user.email'),
+    ]);
+
+    return Boolean(gitUsername.stdout?.trim() && gitEmail.stdout?.trim());
+  } catch (error) {
+    return false;
+  }
+}
+
+/**
+ * Initializes a git repository in the destination folder
+ *
+ * @param dir - source path to initialize git repository in
+ * @param context - template parameters
+ * @throws if `exec` fails
+ */
+export async function initGitRepository(dir: string, context: any) {
+  const runCmd = (cmd: string) =>
+    exec(cmd).catch(error => {
+      process.stdout.write(error.stderr);
+      process.stdout.write(error.stdout);
+      throw new Error(`Could not execute command ${chalk.cyan(cmd)}`);
+    });
+
+  await Task.forItem('init', 'git repository', async () => {
+    process.chdir(dir);
+
+    await runCmd('git init');
+    await runCmd('git commit --allow-empty -m "Initial commit"');
+
+    const defaultBranch = await runCmd(
+      'git branch --format="%(refname:short)"',
+    );
+
+    context.defaultBranch = defaultBranch.stdout?.trim() || 'master';
```

Comment:
This mutation is a bit difficult to track, would rather have this more explicitly passed around as a return value, and merged with the other input for templating

Reply:
Something like this?

```ts
let defaultBranch = '';
if (isGitConfigured) {
     Task.section('Initializing git repository');
     defaultBranch = await initGitRepository(appDir);
}

Task.section('Preparing files');
await templatingTask(templateDir, opts.path, { ...answers, defaultBranch });
```

Reply:
At a high level you could aim for a structure like this, along with the suggestion above in the other thread

```ts
const gitConfig = Task.forItem('git', 'read config', () => readGitConfig());

// ...

Task.section('Preparing files');
await templatingTask(templateDir, opts.path, {
  ...answers,
  defaultBranch: gitConfig?.defaultBranch ?? DEFAULT_GIT_BRANCH,
});
```

If git isn't available then `gitConfig` would be `undefined`, and if that's the case we skip the git init.